library/std/src/sys/unix/fs.rs

```diff
@@ -655,6 +655,11 @@ impl OpenOptions {
     pub fn mode(&mut self, mode: u32) {
         self.mode = mode as mode_t;
     }
+    pub fn as_flags(&self) -> io::Result<c_int> {
+        let access_mode = self.get_access_mode()?;
+        let creation_mode = self.get_creation_mode()?;
+        Ok(creation_mode | access_mode | self.custom_flags)
```

Comment:
Why not just make use of `self.mode`?

Reply:
because  If you say .read(true).write(true). is self.mode reflecting these changes? I doesn't look like it from the code